internal/services/repository/resource_repository_npm_group.go

```diff
@@ -0,0 +1,127 @@
+package repository
+
+import (
+	nexus "github.com/datadrivers/go-nexus-client/nexus3"
+	"github.com/datadrivers/go-nexus-client/nexus3/schema/repository"
+	"github.com/datadrivers/terraform-provider-nexus/internal/schema/common"
+	repositorySchema "github.com/datadrivers/terraform-provider-nexus/internal/schema/repository"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceRepositoryNpmGroup() *schema.Resource {
+	return &schema.Resource{
+		Description: "Use this resource to create a group npm repository.",
+
+		Create: resourceNpmGroupRepositoryCreate,
+		Delete: resourceNpmGroupRepositoryDelete,
+		Exists: resourceNpmGroupRepositoryExists,
+		Read:   resourceNpmGroupRepositoryRead,
+		Update: resourceNpmGroupRepositoryUpdate,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Common schemas
+			"id":     common.ResourceID,
+			"name":   repositorySchema.ResourceName,
+			"online": repositorySchema.ResourceOnline,
+			// Group schemas
+			"group":   repositorySchema.ResourceGroup,
```

Comment:
please use repositorySchema.ResourceGroupDeploy

Reply:
if you don't have a nexus pro license, i can implement this after merging your pull request.

Reply:
extract into new ticket